ui/app/selectors/selectors.js

```diff
@@ -14,6 +14,16 @@ export function getNetworkIdentifier (state) {
   return nickname || rpcUrl || type
 }
 
+export function getCurrentNetworkName (state) {
+  const { metamask: { provider } } = state
```

Comment:
Personal preference: I don't like nested destructuring assignment; I find it hard to read. So, I will suggest the change, but not that it's optional. In this case, it _would_ make it more consistent with `getCurrentChainId` below:
```suggestion
  const { provider } = state.metamask
```